src/Scorecard.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+class Scorecard {
+
+  constructor(){
+    this.frames = 1
+    this.frame_score = []
+    this.scores = []
+    this.running_score = []
+    this.roll_num = 1
+    this.bonus_roll = false
+  }
+
+  add(pins) {
+    if (this.frames < 10 && this.roll_num === 1) {
+      this.getRoll(pins);
+      if (this.isStrike()){
+        this.frames += 1;
+        this.clearFrame();
+      } else {
+        this.roll_num += 1;
+      }
+    } else if (this.frames < 10 && this.roll_num === 2) {
+      this.getRoll(pins);
+      this.frames += 1;
+      this.clearFrame();
+      this.roll_num -= 1;
+    } else if (this.frames === 10 && this.roll_num === 1) {
+      this.getTenthFrameRoll(pins);
+    } else if (this.frames === 10 && this.roll_num === 2) {
+      this.getTenthFrameRoll(pins);
+    } else if (this.frames === 10 && this.roll_num === 3 && this.isBonusBall() === true) {
+      this.getTenthFrameRoll(pins);
+    } else {
+      return this.gameOver();
+      }
+    }
+
+  getRoll(pins){
+    this.frame_score.push(pins);
+    this.scores.push(pins);
+  }
+
+  getTenthFrameRoll(pins){
+    this.getRoll(pins);
+    this.roll_num += 1;
+  }
+
+  isStrike(){
+    return (this.frame_score[0] === 10) ? true : false
+  }
+
+  isSpare(){
+    return (this.frame_score[0] + this.frame_score[1]) === 10 ? true : false
+  }
+
+  isBonusBall(){
+    return (this.frame_score[0] + this.frame_score[1]) >= 10 ? true : false
```

Comment:
You created alot of functions with small amounts of code in them which is good way of keeping code clean. You followed SRP for the most functions.